RecoVertex/Configuration/python/RecoVertex_phase2_timing_cff.py

```diff
@@ -1,20 +1,15 @@
 import FWCore.ParameterSet.Config as cms
 from RecoVertex.Configuration.RecoVertex_cff import unsortedOfflinePrimaryVertices, trackWithVertexRefSelector, trackRefsForJets, sortedPrimaryVertices, offlinePrimaryVertices, offlinePrimaryVerticesWithBS,vertexrecoTask
 
+from RecoVertex.PrimaryVertexProducer.TkClusParameters_cff import DA2D_vectParameters
+
 unsortedOfflinePrimaryVertices4D = unsortedOfflinePrimaryVertices.clone(
-    TkClusParameters = dict(
-        algorithm = "DA2D_vect", 
-        TkDAClusParameters = dict(
-            Tmin = 4.0, 
-            Tpurge = 4.0, 
-            Tstop = 2.0
-        ),
-    ),
+    TkClusParameters = DA2D_vectParameters.clone(),
```

Comment:
Just to say out loud a possible alternative (I'm not really sure if it would be better or worse), if the intention is to use the `DA2D_vect` algorithm with its default options, would
```suggestion
    TkClusParameters = cms.PSet(algorithm = cms.string("DA2D_vect")),
```
work? In principle I'd expect the validation of the `TkDAClusParameters` to insert the missing parameters with their default values.

Reply:
technically I expect it should work identically if the ```fillDescription``` and the python one coincide, because I understand that in this case whatever is missing is taken from the ```fillDescription default```. This is not the case at present. But the same argument could apply to the Run3 default ```DA_vect```, am I correct?

Reply:
@makortel there are a few parameters to be updated, this could be done in the python as well. Still the missing parameters are missing in the python configuration unrolling. I understand that in the code they are taken from the ```fillDescription``` default, I will cross check it just to be sure, but this would make the configurations less readable IMHO.

Reply:
> I understand that in the code they are taken from the `fillDescription` default, I will cross check it just to be sure, but this would make the configurations less readable IMHO.

Ok, thanks for testing it out anyway. Having everything spelled out in the python, rather than relying the configuration validation to inject the missing parameters, is widely spread pattern anyway.

I'd think it would be good to have the python-side and C++-side defaults to match in any case.

Reply:
right, the point is that the same ```TkDAClusParameters``` was originally used for both the Run1-2-3 configuration, and for the 4D extension at HL-LHC, where some values are changed, and others are further added. So the default ```fillDescription``` coincides with the python definition in the former case, i.e. ```DA_vect```, while  some update is needed for ```DA2D_vect```. The possible (working) solution along your suggestion that I have tested would be:
```suggestion
-from RecoVertex.PrimaryVertexProducer.TkClusParameters_cff import DA2D_vectParameters
+# from RecoVertex.PrimaryVertexProducer.TkClusParameters_cff import DA2D_vectParameters
 
 unsortedOfflinePrimaryVertices4D = unsortedOfflinePrimaryVertices.clone(
-    TkClusParameters = DA2D_vectParameters.clone(),
+    TkClusParameters = cms.PSet(algorithm = cms.string("DA2D_vect"),
+        TkDAClusParameters = cms.PSet(
+            Tmin = cms.double(4.0),
+            Tpurge = cms.double(4.0),
+            Tstop = cms.double(2.0),
+        )
+    ),
```
and simultaneously drop the ```DA2D_vectParameters``` definition in https://github.com/cms-sw/cmssw/pull/43846/files#diff-320f7bbd27b83edbb7af9a82948ed917852c026572d6a23d34a2a146dd95b619R55

Reply:
one more remark: not explicitly defining the parameter in the python cfi may work, but it prevents a modifier's ```toModify``` function to update it, in case needed